### PR TITLE
feat: eccentricity, diameter, and as-matrix methods in Graph

### PR DESCRIPTION
**Pull request by nishaq503**
feat: usage of new methods in CHAODA

We now sit on 6 CHAODA methods, potentially 7 after today's meeting.
Later, I might add a faster estimate for Graph.diameter

### CODE REVIEW

File: pyclam/chaoda.py [hunk 1]
```diff
@@ -217,4 +216,23 @@ def random_walk(graph: Graph, normalize: str = 'gaussian') -> np.array:
     return _normalize(np.array([scores[i] for i in range(graph.population)], dtype=int), mode=normalize)
 
 
+def stationary_probabilities(graph: Graph, normalize: str = 'gaussian') -> np.array:
+    scores: Dict[Cluster, float] = {cluster: -1 for cluster in graph.clusters}
```
Comment (thoward27) [inline pyclam/chaoda.py:220]: docstring

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "clam"
-version = "0.2.10"
```
Comment (thoward27) [inline Cargo.toml:3]: I just realized, we don't need to update the rust toml with pyclam, as the python code has no way of being included in the eventual rust crate. We could reset this to 0.1.0 and tie bumps to this to correspond with actual rust releases. The version numbers would be misaligned, but I think that's more informative then this linking we've done so far.

Comment (thoward27) [reply Cargo.toml:3]: Basically, I'd argue the python and rust implementations *should* be on different version numbers to indicate the relative progress both packages have had

File: pyclam/chaoda.py [hunk 3]
```diff
@@ -88,19 +88,18 @@ def graph_neighborhood(graph: Graph, normalize: str = 'gaussian') -> np.array:
     """
     _catch_normalization_mode(normalize)
 
-    def _bft(start: Cluster) -> List[int]:
-        path_length: List[int] = list()
+    def _bft(start: Cluster, steps: int) -> int:
```
Comment (thoward27) [inline pyclam/chaoda.py:91]: seems odd to require a set amount of steps for bft

**Review by thoward27 — CHANGES_REQUESTED** [review]
requesting changes because of a missing docstring on a public function. Other than that I had a thought about our versioning scheme and a nit

File: Cargo.toml (hunk 2 above)
Comment (nishaq503) [reply Cargo.toml:3]: I agree with this. Let's start this with the CHESS branch.

File: pyclam/chaoda.py (hunk 3 above)
Comment (nishaq503) [reply pyclam/chaoda.py]: it returns the number of clusters within a graph distance of 'steps' from 'start'. I changed the name and added a docstring

File: pyclam/chaoda.py (hunk 1 above)
Comment (nishaq503) [reply pyclam/chaoda.py]: done

**Review by thoward27 — APPROVED** [review]
LGTM

**Comment by thoward27** [comment]
@nishaq503 I still think this PR should *reset* the cargo.toml version to 0.1.0 to indicate the progress we've made there

**Pull request merged**